Updating DumpMachine to reflect new output. (#431)
MicrosoftDocs/quantum-docs-pr: articles/techniques/testing-and-debugging.md

Initially this file contains one sample unit test `AllocateQubitTest` which checks that a newly allocated qubit is in the $\ket{0}$ state and prints a message:

```qsharp
    operation AllocateQubitTest () : Unit {
        using (q = Qubit()) {

>             Assert([PauliZ], [q], Zero, "Newly allocated qubit must be in |0> state");

Reviewer: ```suggestion
            Assert([PauliZ], [q], Zero, "Newly allocated qubit must be in the |0⟩ state.");
```

calling <xref:microsoft.quantum.diagnostics.dumpmachine> generates this output:

```
# wave function for qubits with ids (least to most significant): 0;1
∣0❭:	 0.707107 +  0.000000 i	 == 	**********           [ 0.500000 ]     --- [  0.00000 rad ]
∣1❭:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
∣2❭:	-0.500000 + -0.500000 i	 == 	**********           [ 0.500000 ]   /     [ -2.35619 rad ]
∣3❭:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
```

The first row provides a comment with the IDs of the corresponding qubits in their significant order.

> The rest of the rows describe the probability amplitude of measuring the basis state vector $\ket{n}$ in both, cartesian and polar, formats. In detail for the first row:

Reviewer: ```suggestion
The rest of the rows describe the probability amplitude of measuring the basis state vector $\ket{n}$ in both Cartesian and polar formats. In detail for the first row:
```

> * **`∣0❭:`** this rows correspond to the `0` state vector

Reviewer: ```suggestion
* **`∣0❭:`** this row corresponds to the `0` computational basis state
```

> * **`0.707107 +  0.000000 i`**: the probability amplitude in cartesian format.

Reviewer: ```suggestion
* **`0.707107 +  0.000000 i`**: the probability amplitude in Cartesian format.
```

* **` == `**: the `equal` sign seperates both equivalent representations.
* **`**********  `**: A graphical representation of the magnitude, the number of `*` is proportionate to the probability of measuring this state vector.
* **`[ 0.500000 ]`**: the numeric value of the magnitude
* **`    ---`**: A graphical representation of the amplitude's phase (see below).
* **`[ 0.0000 rad ]`**: the numeric value of the phase (in radians).

Both the magnitude and the phase are displayed with a graphical representation. The magnitude representation is straight-forward: it shows a bar of `*`, the bigger the probability the bigger the bar will be. For the phase, we show the following symbols to represent the angle based on ranges:

> ```

Reviewer: This might be easier to read as a DFM table, perhaps?

Author: Maybe later.
